### ISSUE TITLE
error: imported package is not shared

### ISSUE THREAD

**Issue by bjansen** [ceylon/ceylon-compiler#2221]
Adapted from #2024 

Create an empty project with a new module named `plop`.

module.ceylon:
```
module plop "1.0.0" {
    import ceylon.time "1.1.1";
}
```

run.ceylon:
```
import ceylon.time {
    Instant
}

"Run the module `plop`."
shared void run() {
    Instant(0);
}
```

Foo.java:
```
package plop;

public class Foo {
    ceylon.time.Instant instant = null;
}
```

The IDE doesn't show any error, but the CLI throws the following error:
```
mbp:Coin bastien$ ~/Dev/ceylon/ceylon-dist/dist/bin/ceylon compile plop
warning: Package class ceylon.time contains no name, ignoring it
source/plop/run.ceylon:1: error: imported package is not shared: 'ceylon.time'
import ceylon.time {
       ^
1 error
1 warning
Note: Created module plop/1.0.0
ceylon compile: There was 1 error
```

In the Java file, changing the fully qualified reference to an import solves the problem. In my case, Java files are generated by jooq, which uses fully qualified names instead of imports, so I can't change the generated code.

**Comment by FroMage** [comment]
So, this works:

```java
    ceylon.time.Instant instant = new ceylon.time.Instant(0);
```

But not your version with `= null`. Crazy.

**Comment by bjansen** [comment]
They are used as return types:

```
public final org.jooq.TableField<FooRecord, ceylon.time.DateTime> LAST_CHECKED_DATE = createField("last_checked_date", org.jooq.impl.SQLDataType.TIMESTAMP, this, "", new MyDateTimeConverter());
```

**Comment by FroMage** [comment]
OK, should be fixed.

**Comment by FroMage** [comment]
Fixed again @bjansen even for single-part packages this time.
